test/oranif_eunit.erl

```diff
@@ -0,0 +1,1122 @@
+-module(oranif_eunit).
+-include_lib("eunit/include/eunit.hrl").
+
+-define(DPI_MAJOR_VERSION, 3).
+-define(DPI_MINOR_VERSION, 0).
+
+-define(EXEC_STMT(_Conn, _Sql),
+    (fun() ->
+        __Stmt = dpiCall(Safe, conn_prepareStmt, [_Conn, false, _Sql, <<"">>]),
+        catch dpiCall(Safe, stmt_execute, [__Stmt, []]),
+        catch dpiCall(Safe, stmt_release, [__Stmt])
+    end)()
+).
+
+%% gets a value out of a fetched set, compares it using an assertation,
+%% then cleans is up again
+assert_getQueryValue(Safe, Stmt, Index, Value) ->
+    #{data := QueryValueRef} = dpiCall(Safe, stmt_getQueryValue, [Stmt, Index]),
+    ?assertEqual(Value, dpiCall(Safe, data_get, [QueryValueRef])),
+	dpiCall(Safe, data_release, [QueryValueRef]),
+    ok.
+
+
+assert_getQueryInfo(Safe, Stmt, Index, Value, Atom) ->
+    QueryInfoRef = dpiCall(Safe, stmt_getQueryInfo, [Stmt, Index]),
+    ?assertEqual(Value, maps:get(Atom, dpiCall(Safe, queryInfo_get, [QueryInfoRef]))),
+	dpiCall(Safe, queryInfo_delete, [QueryInfoRef]),
+    ok.
+
+extract_getQueryValue(Safe, Stmt, Index) ->
+    #{data := QueryValueRef} = dpiCall(Safe, stmt_getQueryValue, [Stmt, Index]),
+    Result = dpiCall(Safe, data_get, [QueryValueRef]),
+	dpiCall(Safe, data_release, [QueryValueRef]),
+    Result.
+
+extract_getQueryInfo(Safe, Stmt, Index, Atom) ->
+    QueryInfoRef = dpiCall(Safe, stmt_getQueryInfo, [Stmt, Index]),
+    Result = maps:get(Atom, dpiCall(Safe, queryInfo_get, [QueryInfoRef])),
+	dpiCall(Safe, queryInfo_delete, [QueryInfoRef]),
+    Result.
+
+
+iozip(List) -> lists:zip(List, lists:seq(1, length(List))).
+
+simple_fetch({Safe, _Context, Conn}) ->
+    SQL = <<"select 12345, 2, 4, 8.5, 'miau' from dual">>,
+    Stmt = dpiCall(Safe, conn_prepareStmt, [Conn, false, SQL, <<"">>]),
+    Query_cols = dpiCall(Safe, stmt_execute, [Stmt, []]),
+    dpiCall(Safe, stmt_fetch, [Stmt]),
+    #{nativeTypeNum := Type, data := Result} =
+        dpiCall(Safe, stmt_getQueryValue, [Stmt, 1]),
+    ?assertEqual(12345.0, dpiCall(Safe, data_get, [Result])),
+    ?assertEqual(Type, 'DPI_NATIVE_TYPE_DOUBLE'),
+    ?assertEqual(Query_cols, 5),
+    dpiCall(Safe, data_release, [Result]),
+    dpiCall(Safe, stmt_release, [Stmt]).
+
+create_insert_select_drop({Safe, _Context, Conn}) -> 
+    ?EXEC_STMT(Conn, <<"select 12345, 2, 4, 8.5, 'miau' from dual">>),
+    ?EXEC_STMT(Conn, <<"select 12345, 2, 4, 844.5, 'miau' from dual">>),
```

Comment:
What is being tested in this lines ? No side effect so they probably could be removed without changing the result as any error is catch on the Macro `EXEC_STMT` or is my understanding not correct ?

Reply:
@KarlKeiser can please take care of it in your branch?

Reply:
You're right, they're not needed anymore, I removed them (in branch `karl-eunit_cleanup_1`)